src/annotator/sidebar.js

```diff
@@ -53,43 +49,52 @@ function createSidebarIframe(config) {
  * as well as the adjacent controls.
  */
 export default class Sidebar extends Guest {
+  /**
+   * @param {HTMLElement} element
+   * @param {Record<string, any>} config
+   */
   constructor(element, config) {
-    if (config.theme === 'clean' || config.externalContainerSelector) {
-      delete config.pluginClasses.BucketBar;
-    }
-
-    let externalContainer = null;
-
-    if (config.externalContainerSelector) {
-      externalContainer = document.querySelector(
-        config.externalContainerSelector
-      );
-    }
+    super(element, config);
 
     let externalFrame;
     let frame;
+    let hypothesisSidebar;
 
-    if (externalContainer) {
-      externalFrame = externalContainer;
+    if (config.externalContainerSelector) {
+      externalFrame =
+        document.querySelector(config.externalContainerSelector) || element;
     } else {
       frame = document.createElement('div');
       frame.style.display = 'none';
       frame.className = 'annotator-frame annotator-outer';
 
       if (config.theme === 'clean') {
         frame.classList.add('annotator-frame--theme-clean');
+      } else {
+        // BucketBar is a "plugin" for legacy reasons and is now constructed here so
+        // that the parent element can be passed into the constructor.
+        this.plugins.BucketBar = new BucketBar(frame, {}, this);
       }
 
-      element.appendChild(frame);
+      // Undocumented switch to enable/disable the wrapping of the sidebar inside a shadow DOM
+      if (config.disableShadowSidebar) {
+        element.appendChild(frame);
+      } else {
+        // Wrap up the 'frame' element into a shadow DOM so it is not affected by host CSS styles
+        hypothesisSidebar = document.createElement('hypothesis-sidebar');
```

Comment:
Currently this element has no styles and just functions as a container. That means that the browser shows it as having zero size in devtools which is perhaps surprising. Once we're satisfied that the introduction of a Shadow DOM is working and hasn't caused any unexpected issues I think it would make sense to move some of the frame's styles onto this element.

Reply:
In my opinion `<hypothesis-sidebar>` should act purely as a container and should not have associated styles.